dirty_equals/_datetime.py

```diff
@@ -178,3 +178,104 @@ def __init__(
         )
         if tz is not None:
             self._repr_kwargs['tz'] = tz
+
+
+class IsDate(IsNumeric[date]):
+    """
+    Check if the value is a date, and matches the given conditions.
+    """
+
+    allowed_types = date
+
+    def __init__(
+        self,
+        *,
+        approx: Optional[date] = None,
```

Comment:
delta shouldn't be removed as an argument, just the default should change to zero.

Reply:
@samuelcolvin 
I'm working on it. You're right delta should be an option.
I'm not sure which types should be accepted for delta. Personally I think timedelta is enough for type of delta.
Do you think integers should be accepted for delta argument as IsDateTime?
For instance, if delta is passed as 5 then delta is redefined as timedelta(days=5) in __init__ method.